### ISSUE TITLE
[2.1.0-dev.10] question about onvif.media.getProfiles() /onvif.media.getStreamUri(...) 

### ISSUE THREAD

**Issue by LODYZ** [faithoflifedev/easy_onvif#34]
Hi @faithoflifedev , 

i have a question about how to migrate from stable version to the newest. Reading the new documentation I noticed:

```
// defaults to Media2, but will fallback to Media1 if Media2 is not supported
var profiles = await onvif.media.getProfiles();

// alternatively, make an explicit call to the Media2 operation
var profiles2 = await onvif.media.media2.getProfiles();
```

Reading this I expect that no matter what operations the device supports (Media1 or Media2), I'm able to fetch the profiles without needing to know if the device supports Media1 or Media2.

I've tried to call await onvif.media.getProfiles() and also onvif.media.getStreamUri(...) but in both cases I got the same exception: `NotSupportedException` (caused by the fact that the device in question have only media1).

Did I misinterpret the documentation? 

Thank you in advance

PS. Sorry for my English.

**Comment by faithoflifedev** [comment]
HI @LODYZ, it's my error for not being clear in the documentation.

The initial problem that I was trying to address was that the `Profile` response given from a **Media2** device now differs from that of a **Media1** device.   So since `onvif.media.getProfiles()` can only return a single data type I ended up defaulting it to the `MediaProfile` class returned by a **Media2** device, and had it throw a `NotSupportedException` in the case that the device only supports **Media1**.  This causes confusion since all other Onvif Media requests that overlap between Media1 and Meda2 don't really change the response (though some change the parameters in the request).

Overall though, I've made some code changes in the latest release that I think resolves and complication around this issue.  In the v2.1.0-dev.11 release I've created a new class `MixedProfile` that includes all the fields of both a **Media1** and **Media2** response.   Now when you call `onvif.media.getProfiles()` if you're device supports Media1, then all the fields of a `Profile` object are available in the `MixedProfile` response object.  If you're device supports Media2, then all the fields of a `MediaProfile` object are available in the `MixedProfile` response object.  Any of the fields not supported by the device will have null values.

On a Media2 device.  the following would happen:
```dart
var mixedProfiles = await onvif.media.getProfiles();
